### PR TITLE
chore: create separate cf for static site

### PR DESCRIPTION
**Pull request by iamhopaul123**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

**Review by Lou1415926 — APPROVED** [review]

**Review by huanjani — APPROVED** [review]

File: internal/pkg/template/templates/workloads/services/static-site/cf.yml [hunk 1]
```diff
@@ -42,6 +42,67 @@ Resources:
         Rules:
           - ObjectOwnership: BucketOwnerEnforced
 
+  BucketPolicyForCloudFront:
```
Comment (dannyrandall) [inline internal/pkg/template/templates/workloads/services/static-site/cf.yml:45]: can't there only be one bucket policy? maybe this should just be `BucketPolicy` 🤔

File: internal/pkg/manifest/env.go [hunk 2]
```diff
@@ -275,41 +273,14 @@ func (cfg *EnvironmentCDNConfig) UnmarshalYAML(value *yaml.Node) error {
...
 // CDNStaticConfig represents the static config for CDN.
 type CDNStaticConfig struct {
-	Location StaticSiteOrImportedBucket `yaml:"location,omitempty"`
-	Alias    string                     `yaml:"alias,omitempty"`
-	Path     string                     `yaml:"path,omitempty"`
-}
-
-// StaticSiteOrImportedBucket is a custom type which supports unmarshaling yaml which
-// can either be of Static Site service name or an existing S3 bucket URL.
-type StaticSiteOrImportedBucket struct {
```
Comment (efekarakus) [inline internal/pkg/manifest/env.go:285]: 🫡

File: internal/pkg/template/templates/workloads/services/static-site/cf.yml [hunk 3]
```diff
@@ -42,6 +42,67 @@ Resources:
...
+                  - cfDistributionID: !Ref CloudFrontDistribution
+
+  CloudFrontOriginAccessControl:
+    Metadata:
+      'aws:copilot:description': 'Access control to make the content in the S3 bucket only accessible through CloudFront'
+    Type: AWS::CloudFront::OriginAccessControl
+    Properties: 
+      OriginAccessControlConfig: 
+          Description: Access control for static s3 origin
+          Name: !Sub 'copilot-${AppName}-${EnvName}-${WorkloadName}-origin-access-control'
```
Comment (efekarakus) [inline internal/pkg/template/templates/workloads/services/static-site/cf.yml:79]: I couldn't find if there is any limitation on the length of the `Name` field from the docs but would you mind testing out a very long name to figure out the limit? and maybe we can truncate accordingly if needed

File: internal/pkg/template/templates/workloads/services/static-site/cf.yml [hunk 4]
```diff
@@ -42,6 +42,67 @@ Resources:
...
+    Metadata:
+      'aws:copilot:description': 'A CloudFront distribution for global content delivery'
+    Type: AWS::CloudFront::Distribution
+    Properties:
+      DistributionConfig:
+        DefaultCacheBehavior:
+          AllowedMethods: ["GET", "HEAD", "OPTIONS", "PUT", "PATCH", "POST", "DELETE"]
+          ViewerProtocolPolicy: allow-all
+          CachePolicyId: 658327ea-f89d-4fab-a63d-7e88639e58f6 # See https://go.aws/3bJid3k
+          TargetOriginId: !Sub 'copilot-${AppName}-${EnvName}-${WorkloadName}'
```
Comment (efekarakus) [inline internal/pkg/template/templates/workloads/services/static-site/cf.yml:94]: Similarly, is there a length limit here?

**Pull request merged**